imports/plugins/core/router/lib/router.js

```diff
@@ -49,61 +77,109 @@ class Router {
     return Router.routes;
   }
 
+  /**
+   * Triggers reactively on router ready state changed
+   * @return {Boolean} Router initalization state
+   */
   static ready() {
     routerReadyDependency.depend();
     return Router._initialized;
   }
 
+  /**
+   * Re-triggers router ready dependency
+   * @return {undefined}
+   */
   static triggerRouterReady() {
     routerReadyDependency.changed();
   }
 
+  /**
+   * Hooks
+   * @type {Hooks}
+   */
   static get triggers() {
     return Hooks;
   }
 
+  /**
+   * Get the current route date. Not reactive.
+   * @return {Object} Object containing route data
+   */
   static current() {
     return currentRoute.toJS();
   }
 
+  /**
+   * Set current route data. Is reactive.
+   * @param {Object} routeData Object containing route data
+   * @return {undefined}
+   */
   static setCurrentRoute(routeData) {
     currentRoute = Immutable.Map(routeData);
     routerChangeDependency.changed();
   }
 
+  /**
+   * Get the name of the current route. Is reactive.
+   * @return {String} Name of current route
+   */
   static getRouteName() {
     const current = Router.current();
 
     return current.route && current.route.name || "";
   }
 
+  /**
+   * Get param by name. Is reactive.
+   * @param  {String} name Param name
+   * @return {String|undefined} String value or undefined
+   */
   static getParam(name) {
     routerChangeDependency.depend();
     const current = Router.current();
 
     return current.params && current.params[name] || undefined;
   }
 
+  /**
+   * Get query param by name
+   * @param  {String} name Query param name. Is reactive.
+   * @return {String|undefined} String value or undefined
+   */
   static getQueryParam(name) {
     routerChangeDependency.depend();
     const current = Router.current();
 
     return current.query && current.query[name] || undefined;
   }
 
+  /**
+   * Merge new query params with current params
+   * @param {Object} newParams Object containing params
+   * @return {undefined}
+   */
   static setQueryParams(newParams) {
     const current = Router.current();
+
+    // Merge current and new params
     const queryParams = Object.assign({}, current.query, newParams);
 
+    // Any param marked as null or undefined will be removed
     for (const key in queryParams) {
       if (queryParams[key] === null || queryParams[key] === undefined) {
         delete queryParams[key];
       }
     }
 
+    // Update route
     Router.go(current.route.name, current.params, queryParams);
   }
 
+  /**
+   * Watch path change. Is Reactive.
+   * @return {[type]} [description]
```

Comment:
jsdoc can't parse the `[` and `]` inside `{` and `}`s - so, {description}`? what does this method return?